bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/ImageLoader.java

```diff
@@ -149,10 +152,72 @@ void reset() {
  * </ul>
  */
 public ImageData[] load(InputStream stream) {
+	return loadDefault(stream);
+}
+
+/**
+ * Loads an array of <code>ImageData</code> objects from the
+ * specified input stream. If the stream is a SVG File and zoom is not 0,
+ * this method will try to rasterize the SVG.
+ * Throws an error if either an error occurs while loading the images, or if the images are not
+ * of a supported type. Returns the loaded image data array.
+ *
+ * @param stream the input stream to load the images from
+ * @param zoom the zoom factor to apply when rasterizing a SVG.
+ * A value of 0 means that the standard method for loading should be used.
+ * @return an array of <code>ImageData</code> objects loaded from the specified input stream
+ *
+ * @exception IllegalArgumentException <ul>
+ *    <li>ERROR_NULL_ARGUMENT - if the stream is null</li>
+ * </ul>
+ * @exception SWTException <ul>
+ *    <li>ERROR_IO - if an IO error occurs while reading from the stream</li>
+ *    <li>ERROR_INVALID_IMAGE - if the image stream contains invalid data</li>
+ *    <li>ERROR_UNSUPPORTED_FORMAT - if the image stream contains an unrecognized format</li>
+ * </ul>
+ *
+ * @since 3.129
+ */
+public ImageData[] load(InputStream stream, int zoom) {
+	if (stream == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
+    reset();
+    byte[] bytes = null;
+	try {
+		bytes = stream.readAllBytes();
+	} catch (IOException e) {
+		SWT.error(SWT.ERROR_IO, e);
+	}
+	ISVGRasterizer rasterizer = SVGRasterizerRegistry.getRasterizer();
+	if (rasterizer != null && zoom != 0) {
+	    try {
+	    	float scalingFactor = zoom / 100.0f;
+	    	BufferedImage image = rasterizer.rasterizeSVG(bytes, scalingFactor);
+	    	if(image != null) {
+	    		try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+	        	    ImageIO.write(image, "png", baos);
+	        	    try (InputStream in = new ByteArrayInputStream(baos.toByteArray())) {
+	        	    	data = FileFormat.load(in, this);
```

Comment:
This is a quick implementation, but it has some drawbacks. Uses extra memory for the PNG data and for the two conversions (BufferedImage -> PNG -> ImageData). Uses extra CPU time for PNG compression and decompression.

Would be probably better to directly convert `BufferedImage` into `ImageData`.
Don't know whether such utility method exists in Eclipse, but google shows some results:
https://www.google.com/search?client=firefox-b-d&q=awt+bufferedimage+to+swt+imagedata

Reply:
I tried to find a better way for the conversion from `BufferedImage` to `InputStream` but this seems to be the best way. I did not yet try the conversion you are proposing (`BufferedImage` -> `ImageData`). It seems to be much code in the link you provided and I don't know if we need different implementation for different OS as seen in the existing conversions. (PNG -> `ImageData`)

I will try if the solution from the link is working.

Reply:
Update: It seems to work pretty good and also feels faster. I will test this with the future performance tests.

Reply:
> Would be probably better to directly convert BufferedImage into ImageData.
Don't know whether such utility method exists in Eclipse, but google shows some results

I have somewhere posted such thing at eclipse but couldn't find it. And yes it requires a lot of code. The main concern a while back to add such thing was that it creates a dependency to AWT (due to `BufferedImage`) @akurtakov  can maybe tell more.

Reply:
AWT is not a mandatory dependency to SWT and is loaded only by SWT_AWT (https://github.com/eclipse-platform/eclipse.platform.swt/tree/master/bundles/org.eclipse.swt/Eclipse%20SWT%20AWT ) and it is supposed to work by design even without java.desktop module as all functionality should be provided by SWT (wrapping OS and when needed providing small implementation itself). If/whether we want to change this is a bigger question which should not be done here.

Reply:
>  If/whether we want to change this is a bigger question which should not be done here.

pleas be aware of `import java.awt.*` in `SVGRasterizer` then ... so I think this of relates here, and should be done here.

Reply:
> > If/whether we want to change this is a bigger question which should not be done here.
> 
> pleas be aware of `import java.awt.*` in `SVGRasterizer` then ... so I think this of relates here, and should be done here.

I don't think we can get rid of the import in `SVGRasterizer` as JSVG works only with AWT afaik.

Reply:
> I don't think we can get rid of the import in SVGRasterizer as JSVG works only with AWT afaik.

That's what I tried to say, AWT will be a requirement for SWT (SVG) after this change so it would be good to discuss this (possible) issue rater now than later as this was the reason why my previous work for bring BufferedImage>SWT Image into SWT was effectively rejected.

Reply:
You can use it in o.e.swt.svg but not in the other fragments.

Reply:
AWT is fine to be requirement for org.eclipse.swt.svg fragment and whoever decides to depend/include  it. Main project should stick to SWT's Image* API.

Reply:
> AWT is fine to be requirement for org.eclipse.swt.svg fragment and whoever decides to depend/include it. Main project should stick to SWT's Image* API.

Now I understand. So the proposed solution from @DevCharly would be better as we could avoid AWT in SWT.

Reply:
I changed this in ffa17e4b4f65481d700b9d7fc27c305934166080. Now AWT is only used by the new plugin in the class `JSVGRasterizer` (former `SVGRasterizer`).